Vitest harness testing components with QFooter or QHeader produces an exception

<!--

Got a question?
===============
The issue list of this repo is **exclusively** for bug reports and feature requests. Please consider the fact that we cannot support you in the creation of your tests. For such issues regarding implementation, please use the following resources:

- Read the docs: http://quasar-framework.org
- For simple/quick questions ask in the Discord chat: https://discord.gg/5TDhbDg
- For complex questions or requiring help, ask on the forum: http://forum.quasar-framework.org/


Reporting a bug?
================
- Are you sure it isn't already reported? Do a search first! It may have already been answered or even fixed in the development branch (`dev`).

- Are you sure you are reporting to the right repo? If you are not reporting an issue which deals directly with Quasar distributable, then there are [multiple Quasar repos](https://github.com/quasarframework) besides this one:
  -- Quasar CLI: https://github.com/quasarframework/quasar-cli
  -- Quasar Starter Kit (project folder template): https://github.com/quasarframework/quasar-template-default
  -- Electron Wrapper: https://github.com/quasarframework/electron-wrapper
  -- Quasar Play: https://github.com/quasarframework/quasar-play
  -- Documentation Website: https://github.com/quasarframework/quasar-framework.org

- Check if the issue is reproducible with the latest stable version of Quasar CLI. If you are using a pre-release, please indicate the specific version you are using.

- It is **required** that you clearly describe the steps necessary to reproduce the issue you are running into. Issues with no clear repro steps will not be triaged. If an issue labeled "need repro" receives no further input from the issue author for more than 5 days, it will be closed.

- If your issue is resolved but still open, don’t hesitate to close it. In case you found a solution by yourself, it could be helpful to explain how you fixed it.

Have a feature suggestion/request?
=======================
Remove the template from below and provide thoughtful commentary *and code samples* on what this feature means for your product. What will it allow you to do that you can't do today? How will it make current work-arounds straightforward? What potential bugs and edge cases does it help to avoid? etc. Please keep it product-centric.
-->

<!-- BUG REPORT TEMPLATE -->
### Software version

OS: MacBook Pro M1 - v12.4
Node: 16.15.1
NPM: 8.11.0

### What did you get as the error?
An exception:

```
TypeError: Cannot read properties of undefined (reading 'value')
 ❯ ReactiveEffect.fn node_modules/quasar/src/components/footer/QFooter.js:48:23
     46|     const fixed = computed(() =>
     47|       props.reveal === true
     48|       || $layout.view.value.indexOf('F') > -1
       |                       ^
     49|       || ($q.platform.is.ios && $layout.isContainer.value === true)
     50|     )
```

### What were you expecting?
Pass the tests.

### What steps did you take, to get the error?

1. Create components

`src/components/TheHeaderComponent.vue`

```
<template>
  <q-header>
    <q-toolbar>
      <q-toolbar-title> Header </q-toolbar-title>
    </q-toolbar>
  </q-header>
</template>
```

`src/components/TheFooterComponent.vue`

```
<template>
  <q-footer>
    <q-toolbar>
      <q-toolbar-title> Footer </q-toolbar-title>
    </q-toolbar>
  </q-footer>
</template>
```

2. Create unit tests

`test/vitest/__tests__/TheHeaderComponent.spec.js`

```
import { describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'
import { installQuasar } from '@quasar/quasar-app-extension-testing-unit-vitest'
import TheHeaderComponent from 'components/TheHeaderComponent.vue'

installQuasar()

describe('<TheHeaderComponent>', () => {
  it('renders default properly', () => {
    const wrapper = mount(TheHeaderComponent)
    expect(wrapper.exists()).to.be.true
  })
})
```


`test/vitest/__tests__/TheFooterComponent.spec.js`

```
import { describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'
import { installQuasar } from '@quasar/quasar-app-extension-testing-unit-vitest'
import TheFooterComponent from 'components/TheFooterComponent.vue'

installQuasar()

describe('<TheFooterComponent>', () => {
  it('renders default properly', () => {
    const wrapper = mount(TheFooterComponent)
    expect(wrapper.exists()).to.be.true
  })
})

```

3. Run unit test

```
npm run test:unit TheHeaderComponent TheFooterComponent
```

4. Output

```
 FAIL  test/vitest/__tests__/components/TheHeaderComponent.spec.js > <TheHeaderComponent> > renders default properly
TypeError: Cannot read properties of undefined (reading 'value')
 ❯ ReactiveEffect.fn node_modules/quasar/src/components/header/QHeader.js:45:23
     43|     const fixed = computed(() =>
     44|       props.reveal === true
     45|       || $layout.view.value.indexOf('H') > -1
       |                       ^
     46|       || ($q.platform.is.ios && $layout.isContainer.value === true)
     47|     )
 ❯ ReactiveEffect.run node_modules/@vue/reactivity/dist/reactivity.cjs.js:189:25
 ❯ ComputedRefImpl.get value [as value] node_modules/@vue/reactivity/dist/reactivity.cjs.js:1136:39
 ❯ ReactiveEffect.fn node_modules/quasar/src/components/header/QHeader.js:53:17
 ❯ ReactiveEffect.run node_modules/@vue/reactivity/dist/reactivity.cjs.js:189:25
 ❯ ComputedRefImpl.get value [as value] node_modules/@vue/reactivity/dist/reactivity.cjs.js:1136:39
 ❯ ReactiveEffect.getter [as fn] node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:1683:31
 ❯ ReactiveEffect.run node_modules/@vue/reactivity/dist/reactivity.cjs.js:189:25
 ❯ doWatch node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:1814:31
 ❯ Module.watch node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:1661:12


 FAIL  test/vitest/__tests__/components/TheFooterComponent.spec.js > <TheFooterComponent> > renders default properly
TypeError: Cannot read properties of undefined (reading 'value')
 ❯ ReactiveEffect.fn node_modules/quasar/src/components/footer/QFooter.js:48:23
     46|     const fixed = computed(() =>
     47|       props.reveal === true
     48|       || $layout.view.value.indexOf('F') > -1
       |                       ^
     49|       || ($q.platform.is.ios && $layout.isContainer.value === true)
     50|     )
 ❯ ReactiveEffect.run node_modules/@vue/reactivity/dist/reactivity.cjs.js:189:25
 ❯ ComputedRefImpl.get value [as value] node_modules/@vue/reactivity/dist/reactivity.cjs.js:1136:39
 ❯ ReactiveEffect.fn node_modules/quasar/src/components/footer/QFooter.js:62:17
 ❯ ReactiveEffect.run node_modules/@vue/reactivity/dist/reactivity.cjs.js:189:25
 ❯ ComputedRefImpl.get value [as value] node_modules/@vue/reactivity/dist/reactivity.cjs.js:1136:39
 ❯ ReactiveEffect.getter [as fn] node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:1683:31
 ❯ ReactiveEffect.run node_modules/@vue/reactivity/dist/reactivity.cjs.js:189:25
 ❯ doWatch node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:1814:31
 ❯ Module.watch node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:1661:12

Test Files  2 failed (2)
     Tests  2 failed (2)
      Time  27ms

```

5. 'Fixing' the tests

If I remove the QHeader and QFooter components the tests pass:

```
<template>
  <p>Header</p>
</template>
```

`src/components/TheFooterComponent.vue`

```
<template>
  <p>Footer</p>
</template>
```

```
 ✓ test/vitest/__tests__/components/TheHeaderComponent.spec.js (1)
 ✓ test/vitest/__tests__/components/TheFooterComponent.spec.js (1)

Test Files  2 passed (2)
     Tests  2 passed (2)
      Time  1.67s (in thread 30ms, 5569.60%)
```

> You probably need to add `view: { value: 'hhh lpr fff' }`, since Quasar expects that to be a `ref` and access it as `layout.view.value`

Oh it's just I based on `isContainer` property. shouldn't be `isContainer: { value: false }` as well? 😅 

I tried using refs like @jdk2pq but when I run `npm run sync:vitest` it fails with this message repeatly:

```
[5/5] 🔨  Building fresh packages...
$ rimraf dist && tsc --declaration --declarationDir dist/types
../../node_modules/@vue/runtime-core/dist/runtime-core.d.ts:1960:9 - error TS2451: Cannot redeclare block-scoped variable 'defineProps'.

1960   const defineProps: _defineProps
             ~~~~~~~~~~~

  node_modules/@vue/runtime-core/dist/runtime-core.d.ts:2014:9
    2014   const defineProps: _defineProps
                 ~~~~~~~~~~~
    'defineProps' was also declared here.

../../node_modules/@vue/runtime-core/dist/runtime-core.d.ts:1961:9 - error TS2451: Cannot redeclare block-scoped variable 'defineEmits'.

1961   const defineEmits: _defineEmits
             ~~~~~~~~~~~

```

So we have to transform all `ref()` to `{value: xxx}`?

I see there is no PR yet, so I'll do it with @jdk2pq help :D

--- a/packages/unit-vitest/src/helpers/layout-injections.ts
+++ b/packages/unit-vitest/src/helpers/layout-injections.ts
@@ -1,3 +1,5 @@
+import { vi } from 'vitest';
+
 /**
  * Injections for Components with a QPage root Element
  */
@@ -11,7 +13,16 @@ export function qLayoutInjections() {
       right: { size: 300, offset: 0, space: false },
       footer: { size: 0, offset: 0, space: false },
       left: { size: 300, offset: 0, space: false },
-      isContainer: false,
+      isContainer: { value: false },
+      view: { value: 'lHh Lpr lff' },
+      rows: { value: { top: 'lHh', middle: 'Lpr', bottom: 'lff' } },
+      height: { value: 900 },
+      instances: {},
+      update: vi.fn(),
+      animate: vi.fn(),
+      totalWidth: { value: 1200 },
+      scroll: { value: { position: 0, direction: 'up' } },
+      scrollbarWidth: { value: 125 },
     },
   };
 }

--- a/packages/unit-vitest/src/helpers/layout-injections.ts
+++ b/packages/unit-vitest/src/helpers/layout-injections.ts
@@ -1,4 +1,5 @@
 import { vi } from 'vitest';
+import { ref } from 'vue';
 
 /**
  * Injections for Components with a QPage root Element
@@ -13,16 +14,16 @@ export function qLayoutInjections() {
       right: { size: 300, offset: 0, space: false },
       footer: { size: 0, offset: 0, space: false },
       left: { size: 300, offset: 0, space: false },
-      isContainer: { value: false },
-      view: { value: 'lHh Lpr lff' },
-      rows: { value: { top: 'lHh', middle: 'Lpr', bottom: 'lff' } },
-      height: { value: 900 },
+      isContainer: ref(false),
+      view: ref('lHh Lpr lff'),
+      rows: ref({ top: 'lHh', middle: 'Lpr', bottom: 'lff' }),
+      height: ref(900),
       instances: {},
       update: vi.fn(),
       animate: vi.fn(),
-      totalWidth: { value: 1200 },
-      scroll: { value: { position: 0, direction: 'up' } },
-      scrollbarWidth: { value: 125 },
+      totalWidth: ref(1200),
+      scroll: ref({ position: 0, direction: 'up' }),
+      scrollbarWidth: ref(125),
     },
   };
 }

--- a/packages/unit-vitest/tsconfig.json
+++ b/packages/unit-vitest/tsconfig.json
@@ -5,7 +5,8 @@
     "module": "esnext",
     // Needed to avoid errors with vitest @types/chai dep, since it uses `{ default as chai }` on a CommonJS module
     // See https://github.com/vitest-dev/vitest/issues/1076
-    "esModuleInterop": true
+    "esModuleInterop": true,
+    "skipLibCheck": true,
   },
   "include": ["src/helpers"]
 }

--- a/test-project-vite/test/vitest/__tests__/LayoutComponent.test.ts
+++ b/test-project-vite/test/vitest/__tests__/LayoutComponent.test.ts
@@ -1,0 +1,13 @@
+import { installQuasar } from '@quasar/quasar-app-extension-testing-unit-vitest';
+import { mount } from '@vue/test-utils';
+import { describe, expect, it } from 'vitest';
+import LayoutComponent from './demo/LayoutComponent.vue';
+
+installQuasar();
+
+describe('layout example', () => {
+  it('should mount component properly', () => {
+    const wrapper = mount(LayoutComponent);
+    expect(wrapper.exists()).to.be.true;
+  });
+});

--- a/test-project-vite/test/vitest/__tests__/demo/LayoutComponent.vue
+++ b/test-project-vite/test/vitest/__tests__/demo/LayoutComponent.vue
@@ -1,0 +1,12 @@
+<template>
+  <q-header>
+    <q-toolbar>
+      <q-toolbar-title>Header</q-toolbar-title>
+    </q-toolbar>
+  </q-header>
+  <q-footer>
+    <q-toolbar>
+      <q-toolbar-title>Footer</q-toolbar-title>
+    </q-toolbar>
+  </q-footer>
+</template>

